src/csagent/utils.py

```
@@ -2,20 +2,40 @@
 from langchain_core.tools import tool
 from langgraph.config import get_stream_writer
 
 logger = logging.getLogger(__name__)
 
 
+def find_project_root():
+    """Find project root by looking for pyproject.toml."""
+    current = Path(__file__).resolve().parent
+    # print(f"Starting from: {current}")
+
+    for parent in [current] + list(current.parents):
```

Ruff (0.14.13)

Consider `[current, *list(current.parents)]` instead of concatenation

Replace with `[current, *list(current.parents)]`

(RUF005)